it/uk/gov/hmrc/tai/integration/TaxAccountSummarySpec.scala

```diff
@@ -72,20 +113,56 @@ class TaxAccountSummarySpec extends IntegrationSpec {
       }
 
       "throws an InternalServerException when the NPS iabds API returns an INTERNAL_SERVER_ERROR" in {
+        val nino = new Generator(new Random).nextNino
+        val npsIabdsUrl = s"/nps-hod-service/services/nps/person/$nino/iabds/$year"
+
+        val npsTaxAccountUrl = s"/nps-hod-service/services/nps/person/$nino/tax-account/$year"
+        server.stubFor(get(urlEqualTo(npsTaxAccountUrl)).willReturn(ok(taxAccountJson)))
+        val npsEmploymentUrl = s"/nps-hod-service/services/nps/person/$nino/employment/$year"
+        server.stubFor(get(urlEqualTo(npsEmploymentUrl)).willReturn(ok(employmentJson)))
+
+        val apiUrl = s"/tai/$nino/tax-account/$year/summary"
+        val request = FakeRequest(GET, apiUrl)
+          .withHeaders(HeaderNames.xSessionId -> generateSessionId)
+          .withHeaders(HeaderNames.authorisation -> bearerToken)
         server.stubFor(get(urlEqualTo(npsIabdsUrl)).willReturn(aResponse().withStatus(INTERNAL_SERVER_ERROR)))
 
         val result = route(fakeApplication(), request)
         result.map(_.failed.futureValue mustBe a[InternalServerException])
       }
 
       "throws a HttpException when the NPS iabds API returns a SERVICE_UNAVAILABLE" in {
+        val nino = new Generator(new Random).nextNino
+        val npsIabdsUrl = s"/nps-hod-service/services/nps/person/$nino/iabds/$year"
+
+        val npsTaxAccountUrl = s"/nps-hod-service/services/nps/person/$nino/tax-account/$year"
+        server.stubFor(get(urlEqualTo(npsTaxAccountUrl)).willReturn(ok(taxAccountJson)))
+        val npsEmploymentUrl = s"/nps-hod-service/services/nps/person/$nino/employment/$year"
+        server.stubFor(get(urlEqualTo(npsEmploymentUrl)).willReturn(ok(employmentJson)))
+
+        val apiUrl = s"/tai/$nino/tax-account/$year/summary"
+        val request = FakeRequest(GET, apiUrl)
+          .withHeaders(HeaderNames.xSessionId -> generateSessionId)
+          .withHeaders(HeaderNames.authorisation -> bearerToken)
         server.stubFor(get(urlEqualTo(npsIabdsUrl)).willReturn(aResponse().withStatus(SERVICE_UNAVAILABLE)))
 
         val result = route(fakeApplication(), request)
         result.map(_.failed.futureValue mustBe a[HttpException])
       }
 
       "throws a HttpException when the NPS iabds API returns a IM_A_TEAPOT" in {
+        val nino = new Generator(new Random).nextNino
```

Comment:
This block of code is repeated a few times in here - might be nicer to put into a private method